tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/TeamMailboxMemberSetMethod.scala

```diff
@@ -0,0 +1,118 @@
+      .map(InvocationWithContext(_, invocation.processingContext))
+
+  private def setTeamMailboxMemberResponse(username: Username, request: TeamMailboxMemberSetRequest): SMono[TeamMailboxMemberSetResponse] = {
+    val teamMailboxes = request.update.keys.map(teamMailboxNameDTO => TeamMailbox.fromString(teamMailboxNameDTO.value).toOption)
+      .filter(maybeTeamMailbox => maybeTeamMailbox.nonEmpty)
+      .map(maybeTeamMailbox => maybeTeamMailbox.get)
+    val invalidTeamMailboxNameFailureResult = request.update.keys.map(teamMailboxName => TeamMailbox.fromString(teamMailboxName.value) match {
+      case Left(_) => Some(TeamMailboxMemberSetFailure(teamMailboxName, SetError.invalidPatch(SetErrorDescription(s"Invalid teamMailboxName ${teamMailboxName.value}"))))
+      case Right(_) => None
+    }).filter(option => option.nonEmpty)
+      .map(option => TeamMailboxMemberSetResult(Option.empty, option))
+
+    SFlux.fromIterable(teamMailboxes)
+      .flatMap(teamMailbox => updateTeamMailboxMembers(username, teamMailbox, request.update(TeamMailboxNameDTO(teamMailbox.asString()))))
+      .collectSeq()
+      .map(teamMailboxMemberSetResults => TeamMailboxMemberSetResponse(request.accountId,
+        teamMailboxMemberSetResults.filter(setResult => setResult.updated.nonEmpty)
+          .map(setResult => setResult.updated.get -> "").toMap,
+        invalidTeamMailboxNameFailureResult.concat(teamMailboxMemberSetResults.filter(setResult => setResult.notUpdated.nonEmpty))
+          .map(setResult => setResult.notUpdated.get.teamMailboxName -> setResult.notUpdated.get.error).toMap))
+  }
+
+  private def updateTeamMailboxMembers(username: Username,
+                                   teamMailbox: TeamMailbox,
+                                   mapMemberNameToRole: Map[TeamMailboxMemberName, Option[TeamMailboxMemberRoleDTO]]): SMono[TeamMailboxMemberSetResult] = {
+    SFlux(teamMailboxRepository.listMembers(teamMailbox))
+      .collectMap(member => member.username, member => member)
+      .flatMap(mapExistedMembers => mapExistedMembers.get(username) match {
+          case Some(teamMailboxMember) =>
+            if (TeamMemberRole.ManagerRole.equals(teamMailboxMember.role.value)) {
+              updateTeamMailboxMembers(teamMailbox, mapMemberNameToRole, mapExistedMembers)
+            } else {
+              SMono.just(createTeamMailboxMemberSetResult(teamMailbox, SetErrorDescription(s"Not manager of teamMailbox ${teamMailbox.asString()}")))
+            }
+          case None => SMono.just(createTeamMailboxMemberSetResult(teamMailbox, SetErrorDescription(s"Wrong teamMailboxName ${teamMailbox.asString()}")))
+      })
+  }
+
+  private def updateTeamMailboxMembers(teamMailbox: TeamMailbox,
+                  mapMemberNameToRole: Map[TeamMailboxMemberName, Option[TeamMailboxMemberRoleDTO]],
+                  mapExistedMembers: Map[Username, TeamMailboxMember]): SMono[TeamMailboxMemberSetResult] = {
```

Comment:
Idem naming is VERY confusing. Again the strategy to choose what it represent might qualify. I suggest:

`presentMembers`